File: pset2/readabillity/readability.c
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <stdio.h>



int main()
{

    double letters, words, sentences;
    letters =  sentences =  0.0;
    //inicializate with 1 because last sentece is not computated
    words = 1;

    string text = get_string("Text: ");
    for (int i = 0; i < strlen(text); i++)
    {
        //letters  a to z
        if ((text[i] >= 'a' &&  text[i] <=  'z') || (text[i] >= 'A' &&  text[i] <=  'Z'))
        {
            letters++;
        }
        //wprds define by spaces
        if (text[i] == ' ')
        {
            words++;
        }
        //sentences ? . !
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }
    }
    

    //cast to float
    float index = 0.0588 * ((float) letters / (float) words * 100.0) - 0.296 * (((float) sentences / (float) words) * 100.0) - 15.8;


    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        //round
        printf("Grade %d\n", (int)round(index));
    }

    return 0;
}

File: pset6/readability/readability.py
letters = 0
sentences = 0
words = 1

text = input("Text: ")


for c in text:
    if c.islower() or c.isupper():
        letters += 1.0
    elif c == "." or c == "!" or c == "?":
        sentences += 1.0
    elif c.isspace():
        words += 1.0

L =  letters /  words  * 100.0
S =  sentences /  words  * 100.0
index = (0.0588 * L  - 0.296 * S) - 15.8

index = int(round(index))

if index < 1:
     print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f'Grade {index}')




